README.md

```diff
@@ -218,6 +218,124 @@ func userNotificationCenter(
+
+### Step 1: Register the URL scheme
+
+In order for Apple to route a deep link to your application you need to register a URL scheme in your application's info.plist file. This can be done using the editor that xcode provides in xcode from the Info tab of your project settings (reference screenshot below) or editing the info.plist directly - 
+
+![](example1.png)
+
+The required fields are as following - 
+
+1. **Identifier** - The identifier you supply with your scheme distinguishes your app from others that declare support for the same scheme. To ensure uniqueness, specify a reverse DNS string that incorporates your company’s domain and app name. Although using a reverse DNS string is a best practice, it doesn’t prevent other apps from registering the same scheme and handling the associated links.
+2. **URL schemes** - In the URL Schemes box, specify the prefix you use for your URLs.
+3. **Role** - Since your app will be editing the role select the role as editor
+
+In order to edit the info.plist directly, just fill in your app specific details and paste this in your plist - 
+
+```
+<key>CFBundleURLTypes</key>
+<array>
+	<dict>
+		<key>CFBundleTypeRole</key>
+		<string>Editor</string>
+		<key>CFBundleURLName</key>
+		<string>{your_unique_identifier}</string>
+		<key>CFBundleURLSchemes</key>
+		<array>
+			<string>{your_URL_scheme}</string>
+		</array>
+	</dict>
+</array>
+```
+
+
+### Step 2: Whitelist supported URL schemes
+
+Since iOS 9 Apple has mandated that the URL schemes that you app can open need to also be listed in the info.plist. This is in addition to Step 1 above. Even if your app isn't opening any other apps, you still need to list your app's URL scheme in order for deep linking to work. 
+
+This needs to be done in the info.plist directly - 
+
+```
+<key>LSApplicationQueriesSchemes</key>
+<array>
+	<string>{your custom URL scheme}</string>
+</array>
+```
+
+### Step 3: Implement handling deep links in your app
+
+Steps 1 & 2 set your app up for receiving deep links but now is when you need to figure out how to handle them within your app. 
+
+If you are using UIKit, you need to implement [`application:openURL:options:`](https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1623112-application) in your application's app delegate.
+
+Example: 
+
+```
+func application(
+    _ app: UIApplication,
+    open url: URL,
+    options: [UIApplication.OpenURLOptionsKey : Any] = [:]
+) -> Bool {
+    guard let components = NSURLComponents(url: url, resolvingAgainstBaseURL: true),
+          let host = components.host
+    else {
+       print("Invalid deep linking URL")
+       return false
+    }
+                   
+    print("components: \(components.debugDescription)")
+   
+    // Create the deep link
+    guard let deeplink = DeepLinking(rawValue: host) else {
```

Comment:
Actually I guess I do have a comment: I'd either simplify this, or include the relevant classes so it is clear what this means. e.g. put this enum in the readme, or just make a simpler parsing sample

Reply:
Yeah I like the idea of simplifying this with may be just code to parse the url components and for the rest they can always look at the example app. What do you think?

Reply:
Sounds good to me!

Reply:
Totally missed your reply ajay but yeah i'd say parse the components, maybe println(url) or something. a link to sample app (especially once that code is in this repo) would be great.

Reply:
@jordan-griffin that's a good call and something that I had in mine but since the current example app repo is private I didn't do it.

Once I move the example app here I can (deep)link :P it in.

Reply:
@evan-masseau updated to remove the code that was using the readme and added a TODO to link the sample app once I bring it in. 

Thinking about it, I'll probably just add it to this PR or just create another branch since all that code is already reviewed.